doc/user/layouts/shortcodes/replica-options.html

```diff
@@ -2,5 +2,6 @@
 ------------------------------------|------------|-------------------------------------
 `SIZE`                              | `text`     | The size of the replica. For valid sizes, see [cluster replica sizes](/sql/create-cluster-replica#sizes).
 `AVAILABILITY ZONE`                 | `text`     | If you want the replica to reside in a specific availability zone. You must specify an [AWS availability zone ID] in either `us-east-1` or `eu-west-1`, e.g. `use1-az1`. Note that we expect the zone's ID, rather than its name.
-`INTROSPECTION INTERVAL`         | `interval` | The interval at which to collect introspection data. See [Troubleshooting](/ops/troubleshooting) for details about introspection data. The special value `0` entirely disables the gathering of introspection data. Defaults to `1s`.
+`INTROSPECTION INTERVAL`            | `interval` | The interval at which to collect introspection data. See [Troubleshooting](/ops/troubleshooting) for details about introspection data. The special value `0` entirely disables the gathering of introspection data. Defaults to `1s`.
 `INTROSPECTION DEBUGGING`           | `bool`     | Whether to introspect the gathering of the introspection data. Defaults to false.
+`IDLE ARRANGEMENT MERGE EFFORT`     | `integer`  | The amount of effort the replica should exert on compacting arrangements during idle periods. *This is an unstable option! It may be changed or removed at any time.*
```

Comment:
> This is an unstable option! It may be changed or removed at any time.

Is there anything technical that facilitates this instability? If we wanted to remove it we'd have to write a migration to do so, right? Otherwise we'd fail to boot up?

Reply:
We could just remove the option from parsing/planning without touching the catalog. But yes, if we properly wanted to revert #16908, we'd also need to remove the catalog field. I assume that requires a migration, unless our deserialization would just ignore unknown fields in the serialized data?

What I hope to avoid by declaring this option unstable is that users add it to their automation scripts for whatever reason, and then complain about their scripts breaking if we maybe decide that we don't want to expose this knob after all.

Reply:
Ah, right, I forgot that clusters/cluster replicas are different from all other catalog objects, in that we store a JSON blob describing them, rather than the SQL used to create them!

> What I hope to avoid by declaring this option unstable is that users add it to their automation scripts for whatever reason, and then complain about their scripts breaking if we maybe decide that we don't want to expose this knob after all.

Yep, makes sense. Having the call out in the docs is probably the best we can do here, but experience suggests that no one reads the docs, and folks will rely on this anyway, so we should be prepared to support this option ~indefinitel (even if only as a deprecated option that silently does nothing, or something like that).